athena/Radius.tsx

```diff
@@ -37,38 +37,54 @@ export const RadiusItem = (
   vector,
 });
 
-function isAccessibleBase(map: MapData, unit: Unit, vector: Vector) {
-  if (!map.contains(vector)) {
-    return false;
+const cacheMap = new Map();
+
+function getCostBase(map: MapData, unit: Unit, vector: Vector, index?: number) {
+  const tileInfo = map.maybeGetTileInfo(vector, undefined, index);
+  return tileInfo ? tileInfo.getMovementCost(unit.info) : -1;
+}
+
+function getTransitionCostBase(
+  info: UnitInfo,
+  current: TileInfo,
+  parent: TileInfo,
+) {
+  if (parent.group !== current.group) {
+    return parent.getTransitionCost(info) + current.getTransitionCost(info);
   }
+  return 0;
+}
 
+function isAccessibleBase(map: MapData, unit: Unit, vector: Vector) {
   const unitB = map.units.get(vector);
   if (unitB && map.isOpponent(unitB, unit)) {
     return false;
   }
-
   const building = map.buildings.get(vector);
-  if (building && !building.info.isAccessibleBy(unit.info)) {
-    return false;
-  }
+  return !(building && !building.info.isAccessibleBy(unit.info));
+}
 
-  return true;
+function isAccessible(map: MapData, unit: Unit, vector: Vector) {
+  const key = vector.toJSON();
+  let accessible = cacheMap.get(key);
+  if (accessible != null) {
+    return accessible;
+  }
+  accessible = isAccessibleBase(map, unit, vector);
+  cacheMap.set(key, accessible);
```

Comment:
Two notes on this:
* `vectors` are immutable objects that are all the same, so you can use them directly as keys for maps without stringify-ing them.
* We cannot cache at the module level a map (`MapData` instance) changes after mutations, or we might check the movement radius of different maps.

We need one map per `MapData` instance – and then the question is how we evict the memory since it might be running in a long running process on the client or server.

Reply:
> and then the question is how we evict the memory since it might be running in a long running process on the client or server.

First thing that comes to mind is an LRU cache for each map instance in this case

Reply:
I would assume we could use a WeakMap?

Reply:
As in, a `WeakMap<MapData, Map<Vector, boolean>>`

Reply:
> I would assume we could use a WeakMap?

It's viable, although I suggest a more robust library. Let me know what you think about these latest changes